### PR TITLE
ddl: limit the length of the index name

### PR DESCRIPTION
**Pull request by mccxj**

### CODE REVIEW

**Comment by zimulala** [comment]
Please add unit tests.

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -106,6 +106,13 @@ func checkTooLongTable(table model.CIStr) error {
 	return nil
 }
 
+func checkTooLongIndex(index model.CIStr) error {
+	if len(index.L) > mysql.MaxIndexIdentifierLen {
+		return ErrTooLongIdent.Gen("too long index %s", index)
```
Comment (jackysp) [inline ddl/ddl_api.go]: Is there a suitable error message in errname.go?

Comment (mccxj) [reply ddl/ddl_api.go]: ok. I changed.

**Pull request merged**